### PR TITLE
Update navigation editor menu selection dropdown

### LINKED ISSUE #28863: Navigation Screen: Combine Menu Switcher with Add Menu
The new Navigation screen currently has buttons at the top to switch menu and to create a new menu:

Select Menu and Add New

I think it makes sense to combine these into a single menu:

Switch Menu

Creating a new menu opens a modal with an input to give the new menu a name.

--

For longer lists of menus, or menu names that wrap, the switcher should adjust:

image

### PR DESCRIPTION
**Pull request by talldan**
## Description
Closes #28863

Makes some changes to menu creation:
- Moves the create menu button into the menu selection dropdown
- Move the form for creating menus into a modal

## How has this been tested?
Added an e2e test

1. Open the nav screen
2. Open the switch menu dropdown
3. Create a new menu

## Screenshots 
Screenshot 2021-03-03 at 4 13 09 pm

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/edit-navigation/src/components/header/style.scss [hunk 1]
```diff
@@ -28,3 +28,8 @@
 .edit-navigation-header__add-menu {
 	min-width: 220px;
 }
+
+.edit-navigation-header__menu-selection-dropdown {
+	// Appear below the modal overlay.
+	z-index: 99998;
```
Comment (grzim) [inline packages/edit-navigation/src/components/header/style.scss:34]: shouldn't we use a variable here (added to _z-index.scss)?

File: packages/edit-navigation/src/components/header/index.js [hunk 2]
```diff
@@ -64,50 +70,78 @@ export default function Header( {
...
+						{ ( { onClose } ) => (
+							<>
+								<MenuGroup>
+									<MenuItemsChoice
+										value={ selectedMenuId }
+										onSelect={ ( menuId ) => {
+											onSelectMenu( menuId );
+											onClose();
+										} }
+										choices={ menus.map( ( menu ) => ( {
```
Comment (grzim) [inline packages/edit-navigation/src/components/header/index.js:97]: we can use destructuring here `({id, name})`

File: packages/edit-navigation/src/components/header/index.js [hunk 3]
```diff
@@ -64,50 +70,78 @@ export default function Header( {
...
-								<MenuItemsChoice
-									value={ selectedMenuId }
-									onSelect={ onSelectMenu }
-									choices={ menus.map( ( menu ) => ( {
-										value: menu.id,
-										label: menu.name,
-									} ) ) }
-								/>
-							</MenuGroup>
+						{ ( { onClose } ) => (
```
Comment (grzim) [inline packages/edit-navigation/src/components/header/index.js:88]: I think the code would be easier to read if we could separate this function to another file, or if we define elements props outside of JSX

File: packages/edit-navigation/src/components/header/index.js (hunk 2 above)
Comment (talldan) [reply packages/edit-navigation/src/components/header/index.js]: Sounds good, done in bf139c0

File: packages/edit-navigation/src/components/header/index.js (hunk 3 above)
Comment (talldan) [reply packages/edit-navigation/src/components/header/index.js:75]: I think we also want to re-use this for your other PR (https://github.com/WordPress/gutenberg/pull/29201).

I've moved the menu items into a component `MenuSwitcher` in ac70e42

File: packages/edit-navigation/src/components/header/style.scss (hunk 1 above)
Comment (talldan) [reply packages/edit-navigation/src/components/header/style.scss]: Done in a1dc2f3.

**Comment by draganescu** [comment]
I tested this and the only nitpick is that the input in the new menu modal does not get autofocus. Can that / should that be handled in this PR @talldan ?

**Comment by talldan** [comment]
@draganescu Yeah, I did try this, but it breaks the way focus is supposed to return when closing a modal (e.g. pressing escape). So a focus loss will happen when autofocusing.

The new `useFocusReturn` hook is not quite as capable as the higher order component was. I looked at a possible fix, but I think it may involve a big refactor.

I can create a separate issue for it.

**Review by grzim — APPROVED** [review]

**Pull request merged**